source/fluid_pressure_boundary_conditions/density.cc

```diff
@@ -0,0 +1,118 @@
+/*
+  Copyright (C) 2015 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/fluid_pressure_boundary_conditions/density.h>
+#include <aspect/melt.h>
+#include <utility>
+#include <limits>
+
+
+namespace aspect
+{
+  namespace FluidPressureBoundaryConditions
+  {
+
+    template <int dim>
+    void
+    Density<dim>::
+    fluid_pressure_gradient (
+      const MaterialModel::MaterialModelInputs<dim> &material_model_inputs,
+      const MaterialModel::MaterialModelOutputs<dim> &material_model_outputs,
+      std::vector<Tensor<1,dim> > &output
+    ) const
+    {
+      const MaterialModel::MeltOutputs<dim> *melt_outputs = material_model_outputs.template get_additional_output<MaterialModel::MeltOutputs<dim> >();
+      Assert(melt_outputs!=NULL, ExcMessage("Error, MeltOutputs are missing in fluid_pressure_gradient()"));
+      for (unsigned int q=0; q<output.size(); ++q)
+        {
+          const Tensor<1,dim> gravity = this->get_gravity_model().gravity_vector(material_model_inputs.position[q]);
+          output[q] = (((include_rho_s) ? material_model_outputs.densities[q] : 0.0)
```

Comment:
What happens here if both are true? Is there a reason you use two bools to decide which of the two you use? Do you expect a third option to be added later? Because in that case you could use an enum and a switch-statement.